Text/CustomEntitySearch/CustomEntitySearch.cs

```diff
@@ -95,5 +132,174 @@ public static class CustomEntitySearch
+        {
+            string wordCharArray = (caseSensitive) ? CreateWordArray(checkMatch) : CreateWordArray(checkMatch.ToLower(CultureInfo.CurrentCulture));
+            if (leniency == 0)
+            {
+                // Overlap checker now also included in Regex expression using delineating characters as overlap lookahead
+                StringBuilder escapedWord = new StringBuilder(@"(?=(");
+                if (!wordCharArray.First().IsDelineating() && !substringMatch)
+                    escapedWord.Append(@"\b");
+                for (int currWordCharIndex = 0; currWordCharIndex < wordCharArray.Length; currWordCharIndex++)
+                {
+                    if (wordCharArray[currWordCharIndex].IsDelineating())
+                    {
+                        escapedWord.Append(".");
+                    }
+                    else
+                    {
+                        escapedWord.Append(wordCharArray[currWordCharIndex]);
+                    }
+                }
+                if (!wordCharArray.Last().IsDelineating() && !substringMatch)
+                    escapedWord.Append(@"\b");
+                escapedWord.Append("))");
+                string pattern = (caseSensitive) ? @"(?x)" + escapedWord : @"(?ix)" + escapedWord;
+                
+                MatchCollection entityMatch = Regex.Matches(text, pattern, RegexOptions.Compiled, TimeSpan.FromSeconds(MaxRegexEvalTime));
+                if (entityMatch.Count != 0)
+                {
+                    foreach (Match match in entityMatch)
+                    {
+                        entities.Add(
+                            new Entity
+                            {
+                                Category = "customEntity",
+                                Value = match.Groups[1].Value,
+                                Offset = match.Index,
+                                Confidence = 0
+                            });
+                    }
+                    entitiesFound.Add(checkMatch);
+                }
+            }
+            else
+            {
+                List<int> pointersInText = new List<int>{ 0 };
+                string textCharArray = (caseSensitive) ? text : text.ToLower(CultureInfo.CurrentCulture);
+                for (int currTextCharIndex = 0; currTextCharIndex < textCharArray.Length; currTextCharIndex++)
+                {
+                    if (textCharArray[currTextCharIndex].IsDelineating())
+                        pointersInText.Add(currTextCharIndex + 1);
+                }
+                pointersInText.Add(textCharArray.Length + 1);
+
+                double[] minLevenshteinDistance = new double[pointersInText.Count];
+                int[] endofMatchInTextPointer = new int[pointersInText.Count];
+                if (leniency > wordCharArray.Length - 1)
+                    leniency = Math.Max(0, wordCharArray.Length - 1);
+                for (int pointer = 0; pointer < pointersInText.Count - 1; pointer++)
```

Comment:
>pointer [](start = 25, length = 7)

nit: this is index to the pointer, not the pointer itself. Mayube rename variable to make that clear?